src/modules/terraform.rs

```diff
@@ -63,6 +54,59 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     Some(module)
 }
 
+fn get_cli_version(module: &Module, ctx: &Context, cfg: &TerraformConfig) -> Option<String> {
+    let version = parse_terraform_version(&ctx.exec_cmd("terraform", &["version"])?.stdout)?;
+    format(module, &version, cfg)
+}
+
+fn parse_terraform_version(version: &str) -> Option<String> {
+    // `terraform version` output looks like this
+    // Terraform v0.12.14
+    // With potential extra output if it detects you are not running the latest version
+    let version = version
+        .lines()
+        .next()?
+        .trim_start_matches("Terraform ")
+        .trim()
+        .trim_start_matches('v');
+
+    Some(version.to_string())
+}
+
+fn format(module: &Module, version: &str, cfg: &TerraformConfig) -> Option<String> {
+    VersionFormatter::format_module_version(module.get_name(), version, cfg.version_format)
+}
+
+fn get_tfenv_version(ctx: &Context, module: &Module, cfg: &TerraformConfig) -> Option<String> {
+    ctx.get_env("TFENV_TERRAFORM_VERSION")
+        .or_else(|| try_tfenv_file(ctx))
+        .and_then(parse_tfenv_version)
+        .and_then(|version| format(module, &version, cfg))
+}
+
+fn try_tfenv_file(ctx: &Context) -> Option<String> {
+    let file_path = ctx.current_dir.join(TFENV_VERSION_FILE);
+    if !file_path.exists() || !file_path.is_file() {
+        return None;
+    }
+    utils::read_file(file_path).ok()
+}
+
+fn parse_tfenv_version(version: String) -> Option<String> {
+    // Tfenv versions are simply exact versions starship expects, like 0.6.16 or 1.0.8
+    // so the only thing to be done is to remove newline.
+    // It is possible to specify version like `latest:^0.8` and it's left as is,
+    // because it's not possible to determine the exact version without calling
+    // `terraform` binary.
```

Comment:
This seems fine. It would be nice if we could resolve the exact version, but if it's going to take a second each time, there's no way that's worth it.